packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -183,6 +232,90 @@ export default class UserStorageController extends BaseController<
     },
   };
 
+  #accounts = {
+    // This is replaced with the actual value in the constructor
+    // We will remove this once the feature will be released
+    isAccountSyncingEnabled: false,
+    setIsUserStorageAccountSyncingInProgress: async (
+      isUserStorageAccountSyncingInProgress: boolean,
+    ) => {
+      // Publish event
+      const eventToPublish = isUserStorageAccountSyncingInProgress
+        ? 'UserStorageController:accountSyncingInProgress'
+        : 'UserStorageController:accountSyncingComplete';
+
+      this.messagingSystem.publish(
+        eventToPublish,
+        isUserStorageAccountSyncingInProgress,
+      );
+
+      // Update state
+      this.update((state) => {
+        state.isUserStorageAccountSyncingInProgress =
+          isUserStorageAccountSyncingInProgress;
+      });
+    },
+    getInternalAccountByAddress: async (address: string) => {
+      return this.messagingSystem.call(
+        'AccountsController:getAccountByAddress',
+        address,
+      );
+    },
+    getInternalAccountsList: async (): Promise<InternalAccount[]> => {
+      return this.messagingSystem.call('AccountsController:listAccounts');
+    },
+    getUserStorageAccountsList: async (): Promise<
+      UserStorageAccount[] | null
+    > => {
+      const rawAccountsListResponse =
+        await this.performGetStorageAllFeatureEntries('accounts');
+
+      return (
+        rawAccountsListResponse?.map((rawAccount) => JSON.parse(rawAccount)) ??
+        null
+      );
+    },
+    saveInternalAccountToUserStorage: async (address: string) => {
+      const internalAccount = await this.#accounts.getInternalAccountByAddress(
+        address,
+      );
+
+      if (!internalAccount) {
+        return;
+      }
+
+      // Map the internal account to the user storage account schema
+      const mappedAccount =
+        mapInternalAccountToUserStorageAccount(internalAccount);
```

Comment:
This is nice.

Compared to network syncing, we can map the shape to a small shape with only the relevant info.

I don't foresee the accounts shape changing, so we don't need to handle versioning.

There will be a future item to handle versioning as a whole (alongside network syncing)